development/tools/cmd/synchronizer/main.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	sc "github.com/kyma-project/test-infra/development/tools/cmd/synchronizer/syncomponent"
+	t "github.com/kyma-project/test-infra/development/tools/cmd/synchronizer/tools"
+	"github.com/pkg/errors"
+)
+
+const (
+	envKymaProjectDir  = "KYMA_PROJECT_DIR"
+	slackClientToken   = "SLACK_CLIENT_TOKEN"
+	slackClientChannel = "STABILITY_SLACK_CLIENT_CHANNEL_ID"
+	outOfDate          = "OUT_OF_DATE_DAYS"
+	pathToVersionFile  = "Makefile"
+	varsionPathCommand = "version-path"
+)
+
+// ComponentStorage inludes pack of components
+type ComponentStorage struct {
+	components []*sc.Component
+}
+
+// AddComponent adds new Component to storage
+func (cs *ComponentStorage) AddComponent(comp *sc.Component) {
+	cs.components = append(cs.components, comp)
+}
+
+func main() {
+	rootDir := os.Getenv(envKymaProjectDir)
+	slackToken := os.Getenv(slackClientToken)
+	slackChannel := os.Getenv(slackClientChannel)
+	outOfDateDays, err := strconv.Atoi(os.Getenv(outOfDate))
+
+	if err != nil {
+		log.Printf("Cannot tranform %s env to integer, default value will be used", outOfDate)
+		outOfDateDays = 0
+	}
+	if rootDir == "" {
+		log.Fatalf("missing env: %s", envKymaProjectDir)
+	}
+	if slackToken == "" {
+		log.Fatalf("missing env: %s", slackClientToken)
+	}
+	if slackChannel == "" {
+		log.Fatalf("missing env: %s", slackClientChannel)
+	}
+
+	storage := &ComponentStorage{}
+
+	findComponents(rootDir, storage)
+	fillComponentStorage(rootDir, storage, outOfDateDays)
+
+	reports := sc.GenerateMessage(storage.components)
+	alertAmount := len(reports)
+	log.Printf("There are %d components with alerts \n", alertAmount)
+	if alertAmount == 0 {
+		return
+	}
+
+	messages := make([]t.Message, alertAmount)
+	for i := range reports {
+		messages[i] = reports[i]
+	}
+	err = t.SendMessage(slackToken, strings.Trim(slackChannel, "#"), messages)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func findComponents(dir string, storage *ComponentStorage) {
```

Comment:
why you pass storage as an argument rather returning it?

Reply:
changed